### ISSUE TITLE
`reportMissingTypeArgument` not fired for bare `type`

### ISSUE THREAD

**Issue by sterliakov** [microsoft/pyright#4658]
**Describe the bug**
`reportMissingTypeArgument` error is not triggered when bare `type` is used as annotation.

**To Reproduce**
Type check the following code:

```python
x: list
t: type
```

(configuration: `"reportMissingTypeArgument": "error"` and `include`, no other flags)

**Observed behavior**
```
Loading configuration file at /tmp/foo/pyrightconfig.json
Config "include" entry must must contain an array.
Assuming Python version 3.10
Assuming Python platform Linux
No include entries specified; assuming /tmp/foo
Auto-excluding **/node_modules
Auto-excluding **/__pycache__
Auto-excluding **/.*
stubPath /tmp/foo/typings is not a valid directory.
Searching for source files
Auto-excluding /tmp/foo/venv
Found 1 source file
pyright 1.1.294
/tmp/foo/a.py
  /tmp/foo/a.py:1:4 - error: Expected type arguments for generic class "list" (reportMissingTypeArgument)
1 error, 0 warnings, 0 informations 
Completed in 0.701sec
```
**Expected behavior**
```
...
/tmp/foo/a.py
  /tmp/foo/a.py:1:4 - error: Expected type arguments for generic class "list" 
  /tmp/foo/a.py:2:4 - error: Expected type arguments for generic class "type" 
...
```

**VS Code extension or command-line**
CLI, `pyright 1.1.294`, I won't install VSCode to check there too.

**Additional context**
I'm not 100% sure that this is a bug, because `mypy` does the same. However, the corresponding [bug issue](https://github.com/python/mypy/issues/12301) for `mypy` is open. This is not addressed in PEP 585.

**Comment by erictraut** [comment]
This is as designed. The symbol `type` is a special in that it acts as both a generic and a non-generic type. It has legitimate uses without a type argument.
